### PR TITLE
Plugins create_lr_scheduler support

### LINKED ISSUE #2576: Plugin `create_lr_scheduler` is not used
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Use `create_lr_scheduler` to create custom lr schedulers

### Current behaviour

Ignored and never called

### Steps to reproduce

Define `create_lr_scheduler` in plugin, check if it is called

### Config yaml

```yaml

```

### Possible solution

Use `create_lr_scheduler` in trainer builder if returned value is not None

### Which Operating Systems are you using?

- [x] Linux
- [x] macOS
- [x] Windows

### Python Version

3.11

### axolotl branch-commit

main

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

### PR DESCRIPTION
**Pull request by alexdremov**
# Description

Support `create_lr_scheduler` method for schedulers. This is breaking changes to the interface, but as feature was not working before, there should be no dependencies. 

Fixes https://github.com/axolotl-ai-cloud/axolotl/issues/2576

## Motivation and Context

Support already introduced plugin feature

## How has this been tested?

- Presonal plugin lr_scheduler
- Enabled e2e tests

## Social Handles

@alexdremov everywhere

### CODE REVIEW

**Review by winglian — APPROVED** [review]
🔥 thank you!

**Comment by alexdremov** [comment]
- [x]  Problem with passing correct `cfg`

**Comment by alexdremov** [comment]
> 🔥 thank you!

Had to change something since you've approved. Specifically, we do not have access to initial axolotl cfg inside trainer. Had to add such requirement to the mixin

**Comment by winglian** [comment]
@alexdremov See https://github.com/axolotl-ai-cloud/axolotl/pull/2587

I'm thinking of merging that one first, which should simplify the need for the workaround of setting `.axolotl_cfg` on the mixin

**Comment by alexdremov** [comment]
> @alexdremov See #2587
> 
> I'm thinking of merging that one first, which should simplify the need for the workaround of setting `.axolotl_cfg` on the mixin

perfect
ping me when it's merged

**Comment by winglian** [comment]
We've merged #2587, thanks!

**Comment by winglian** [comment]
I've rebased and updated this PR

**Review by winglian — APPROVED** [review]
